### PR TITLE
Product show

### PR DESCRIPTION
**Pull request by Jbern16**

### CODE REVIEW

File: app/controllers/product_images_controller.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ def new
   end
 
   def create
+    binding.pry
```
Comment (pindell-matt) [inline app/controllers/product_images_controller.rb:9]:
```suggestion

```

File: app/javascript/controllers/gallery_controller.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      image.onload = () => {
+          gallery.currItem.w = image.width;
+          gallery.currItem.h = image.height;
+
+          gallery.updateSize();
+      };
+    });
+
+        gallery.init();
+  }
```
Comment (pindell-matt) [inline app/javascript/controllers/gallery_controller.js:55]: indenting seems a bit off here?

Comment (Jbern16) [reply app/javascript/controllers/gallery_controller.js:55]: prob, I need to find some auto formatters

File: app/controllers/product_images_controller.rb [hunk 3]
```diff
@@ -22,7 +22,7 @@ def product_params
   end
 
   def product_image_params
-    params.require(:product).permit(:images)
+    params.require(:product).permit({variants_attributes: [:images, :id]})
```
Comment (jdliss) [inline app/controllers/product_images_controller.rb:25]: Another small formatting change
```suggestion
    params.require(:product).permit({ variants_attributes: [:images, :id] })
```

File: app/models/product.rb [hunk 4]
```diff
@@ -3,4 +3,23 @@ class Product < ApplicationRecord
...
+  def all_images
+    all = []
+    images.each do |img|
+      all << img
+    end
+    variants.each do |variant|
+      variant.images.each do |img|
+        all << img
+      end
+    end
```
Comment (jdliss) [inline app/models/product.rb:21]: Should be able to refactor this to something like:

```ruby
all = (images + variants.map(&:images)).flatten
```

**Review by jdliss — APPROVED** [review]
LGTM!

**Pull request merged**